pkg/provisioner/ironic/ironic.go

```diff
@@ -1344,7 +1351,7 @@ func (p *ironicProvisioner) Deprovision() (result provisioner.Result, err error)
 	case nodes.CleanFail:
 		if ironicNode.Maintenance {
 			p.log.Info("clearing maintenance flag")
-			return p.setMaintenanceFlag(ironicNode, false)
+			return result, p.setMaintenanceFlag(ironicNode, false)
```

Comment:
This is wrong, you need to *not* return here unless there is an error, so we can carry on and call `changeNodeProvisionState()`. Since we don't set Dirty=true, we'll never get called again after returning from here.